packages/expo-sqlite/src/Storage.ts

```diff
@@ -0,0 +1,392 @@
+import { openDatabaseAsync, openDatabaseSync, type SQLiteDatabase } from './index';
+
+/**
+ * Update function for the [`setItemAsync()`](#setitemasynckey-value) or [`setItemSync()`](#setitemsynckey-value) method. It computes the new value based on the previous value. The function returns the new value to set for the key.
+ * @param prevValue The previous value associated with the key, or `null` if the key was not set.
+ * @returns The new value to set for the key.
+ */
+export type SQLiteStorageSetItemUpdateFunction = (prevValue: string | null) => string;
+
+const DATABASE_VERSION = 1;
+const STATEMENT_GET = 'SELECT value FROM storage WHERE key = ?;';
+const STATEMENT_SET =
+  'INSERT INTO storage (key, value) VALUES (?, ?) ON CONFLICT(key) DO UPDATE SET value = excluded.value;';
+const STATEMENT_REMOVE = 'DELETE FROM storage WHERE key = ?;';
+const STATEMENT_GET_ALL_KEYS = 'SELECT key FROM storage;';
+const STATEMENT_CLEAR = 'DELETE FROM storage;';
+
+const MIGRATION_STATEMENT_0 =
+  'CREATE TABLE IF NOT EXISTS storage (key TEXT PRIMARY KEY NOT NULL, value TEXT);';
+
+/**
+ * Key-value store backed by SQLite. This class accepts a `databaseName` parameter in its constructor, which is the name of the database file to use for the storage.
+ */
+export class SQLiteStorage {
+  private db: SQLiteDatabase | null = null;
+
+  constructor(private readonly databaseName: string) {}
+
+  //#region Asynchronous API
+
+  /**
+   * Retrieves the value associated with the given key asynchronously.
+   */
+  getItemAsync = async (key: string): Promise<string | null> => {
```

Comment:
I would be more in favor of using function syntax over arrow functions. If you don't need to capture `this` from the outer scope, use a function.

Also, assigning an arrow function to a property like this is not adding the function into the prototype, but to each instance (slower object initialization, higher memory consumption).

Reply:
i was thinking if people calling the storage functions in callbacks and the `this` being overrode. maybe it's an unnecessary pre-optimization. anyhow, i've reverted the changes in https://github.com/expo/expo/pull/31676